packages/itwinui-react/src/core/Panels/Panels.tsx

```diff
@@ -0,0 +1,488 @@
+ *   as={Surface}
+ *   style={{ inlineSize: '300px', blockSize: '500px' }}
+ * >
+ *   <Panels.Panel
+ *     id={panelIdRoot}
+ *     as={Surface}
+ *     border={false}
+ *     elevation={0}
+ *   >
+ *     <Surface.Header as={Panels.Header}>Root</Surface.Header>
+ *     <Surface.Body as={List}>
+ *       <ListItem>
+ *         <Panels.Trigger for={panelIdMoreInfo}>
+ *           <ListItem.Action>More details</ListItem.Action>
+ *         </Panels.Trigger>
+ *       </ListItem>
+ *     </Surface.Body>
+ *   </Panels.Panel>
+ *
+ *   <Panels.Panel
+ *     id={panelIdMoreInfo}
+ *     as={Surface}
+ *     border={false}
+ *     elevation={0}
+ *   >
+ *     <Surface.Header as={Panels.Header}>More details</Surface.Header>
+ *     <Surface.Body isPadded>
+ *       <Text>Content</Text>
+ *     </Surface.Body>
+ *   </Panels.Panel>
+ * </Panels.Wrapper>
+ */
+const PanelsWrapper = React.forwardRef((props, forwardedRef) => {
+  const {
+    initialActiveId,
+    children,
+    className,
+    onActiveIdChange,
+    instance: instanceProp,
+    ...rest
+  } = props;
+
+  const instanceBackup = Panels.useInstance();
+  const instance = instanceProp || instanceBackup;
+
+  const ref = React.useRef<HTMLDivElement | null>(null);
+
+  const [activePanel, _setActivePanel] = React.useState(initialActiveId);
+  const setActivePanel = React.useCallback(
+    (newActivePanel: typeof activePanel) => {
+      _setActivePanel(newActivePanel);
+      onActiveIdChange?.(newActivePanel);
+    },
+    [onActiveIdChange],
+  );
+
+  const previousActivePanel = useDelayed(activePanel);
+
+  const [_panelElements, setPanelElements] = React.useState<
+    Record<string, HTMLElement | null>
+  >({});
+  const panelElements = useLatestRef(_panelElements);
+
+  const [triggers, setTriggers] = React.useState<
+    Record<string, TriggerMapEntry>
+  >({});
+  const triggersRef = useLatestRef(triggers);
+
+  const changeActivePanel = React.useCallback(
+    async (newActiveId: string) => {
+      if (
+        // Panel transition in progress
+        activePanel !== previousActivePanel ||
+        // Only if forProp is a panel, go to the new panel.
+        !Object.keys(panelElements.current).includes(newActiveId) ||
+        // Same panel
+        newActiveId === activePanel
+      ) {
+        return;
+      }
+
+      ReactDOM.flushSync(() => setActivePanel(newActiveId));
+      panelElements.current?.[newActiveId]?.scrollIntoView({
+        block: 'nearest',
+        inline: 'center',
+        behavior: 'smooth',
+      });
+
+      await new Promise((resolve) => {
+        setTimeout(() => {
+          const prevElement = document.getElementById(previousActivePanel);
+          const activeElement = document.getElementById(newActiveId);
+
+          if (prevElement != null && activeElement != null) {
+            const isActivePanelAfterPrevPanelInDom =
+              prevElement?.compareDocumentPosition(activeElement) &
+              Node.DOCUMENT_POSITION_FOLLOWING;
```

Comment:
Such hacks should not be necessary at all. I'm not sure what you're comparing the DOM position like this when we already have a dedicated `goBack` method which is separate from the normal trigger.

Reply:
It is well-supported, but it is a hack - this is an API that we should almost never need to use.

Here's how I think about it:
1. When a `Panel.Trigger` is clicked:
   - We should store this trigger element's ID (let's call it `triggeredBy`). This state is only relevant for the lifecycle of the Panel.
   - We move focus into the `Panel.Header`.
2. When `goBack` is called.
   - We move focus to `triggeredBy`. This should work for both `Panel.BackButton` and also for our consumer's custom `goBack` calls (e.g. they might call `goBack` after an element is selected from the second Panel).
   - We clear `triggeredBy`, since the Panel has reached the end of its lifecycle. It can be repopulated when triggered again (by the same trigger or a different trigger).